experimental/personnames/src/derive_name_order.rs

```diff
@@ -0,0 +1,111 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_locid::subtags::Language;
+use icu_locid::Locale;
+use icu_locid_transform::fallback::LocaleFallbacker;
+use zerovec::VarZeroVec;
+
+use crate::api::FormattingOrder;
+
+///
+/// https://www.unicode.org/reports/tr35/tr35-personNames.html#derive-the-name-order
+///
+pub(crate) fn name_order_derive(
+    person_name_locale: &Locale,
+    surname_first: &VarZeroVec<str>,
+    given_first: &VarZeroVec<str>,
+) -> FormattingOrder {
+    // Set up a LocaleFallbacker with data.
+    let fallbacker = LocaleFallbacker::new();
+    // Create a LocaleFallbackerIterator with a default configuration.
+    // By default, uses language priority with no additional extension keywords.
+    let mut fallback_iterator = fallbacker
+        .for_config(Default::default())
+        .fallback_for(person_name_locale.into());
+
+    loop {
+        let chain_locale = fallback_iterator.get().clone().into_locale();
+        let chain_locale_str = chain_locale.to_string();
+
+        // switch lookup with UND
+        let mut chain_locale_und = chain_locale.clone();
+        chain_locale_und.id.language = Language::UND;
+        let chain_locale_und_str = chain_locale_und.to_string();
```

Comment:
Should be able to do this without converting to a string. Can you use either Locale::partial_eq or Locale::cmp_bytes?